stock_picking_product_availability_inline/models/stock_picking.py

```diff
@@ -0,0 +1,32 @@
+# Copyright 2022 Tecnativa - Ernesto Tejeda
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    picking_type_warehouse_id = fields.Many2one(
+        comodel_name="stock.warehouse",
+        string="Stock picking type warehouse",
+        related="picking_type_id.warehouse_id",
+        store=True,
+        readonly=False,
+    )
+
```

Comment:
Why do you not use this field https://github.com/odoo/odoo/blob/b6833702044b0d3057d543dae71fbea7ef091d98/addons/stock/models/stock_picking.py#L50?

Reply:
This field doesn't store the picking type warehouse. Isn't the idea to work with the 'picking type' warehouse?